internal/vpn.go

```diff
@@ -112,11 +113,40 @@ func updateConfig(app *App, client *http.Client, pubkey string) error {
 	return nil
 }
 
+func vpnBugFix(app *App, sotaConfig string) bool {
+	// We had a race condition fixed in commit f4baae04cdf381d0a61a46c1f64c774f54020256
+	// However, we have devices in the field that were already in a bad
+	// state. This logic exists to fix them. Eventually, all those devices
+	// will pick up this OTA, and we'll be able to remove this quirk.
+	secretFile := filepath.Join(app.SecretsDir, "wireguard-client")
+	if _, err := os.Stat(secretFile); os.IsNotExist(err) {
+		// This is a little over-protective and edge--case. But perhaps an
+		// operator wants to delete this config file? We keep flag so that
+		// we only try this hack one time.
+		migrated := filepath.Join(sotaConfig, ".wg-priv.fixed")
+		if _, err := os.Stat(migrated); os.IsNotExist(err) {
+			log.Println("Wireguard key not registered on server, will re-register")
+			if err = ioutil.WriteFile(migrated, []byte{}, 0o600); err != nil {
+				log.Printf("Unable to write flag file: %s", err)
+			}
+			return true
```

Comment:
Can't we still get in a race after returning true here and possibly not being able to do the actual registration?

Reply:
yep. that's not gonna work. i'll back that bit out

Reply:
be quiet @gavingavel  - nobody asked you!

force push removed that buggy logic.